recipes-mender/rcu-state-scripts/files/retain-credentials

```diff
@@ -5,13 +5,8 @@
 
 echo "$(mender show-artifact): Running $(basename "$0")" >&2
 
-# Check if fw_printenv command is available
-if [ ! -x /usr/bin/fw_printenv ]; then
-    exit 1
-fi
-
 # Check current rootfs partition
-current=$(/usr/bin/fw_printenv mender_boot_part | awk -F = '{ print $2 }')
+current=$(cat /proc/cmdline | awk -F' ' '{ gsub("root=/dev/mmcblk0p", ""); print $2 }')
```

Comment:
A bit of info on the awk command. The `-F` specifies the separator, which in this case is a whitespace. `gsub` is then used to delete the `root=/dev/mmcblk0p` string, and $2 is the second group after the separator is applied to /proc/cmdline.

This is an example of the raw output of /proc/cmdline:
```
pci=nomsi root=/dev/mmcblk0p3 ro rootwait
```

so we grab `root=/dev/mmcblk0p3`, remove `root=/dev/mmcblk0p` and is left with `3`

Reply:
is /proc/cmdline available by default on rcu image? just want to confirm that it is something that is already available.

Reply:
@shlooiNI I believe proc is a default part of the Linux kernel, so it should be readily available. Seems like it goes a long way back too...
https://elixir.bootlin.com/linux/latest/source/Documentation/filesystems/proc.rst

Reply:
Not sure how frequent the command line is changed. If you want a more consistent output, maybe can try `mount`, e.g. `mount | awk '$3 == "/" {sub(/\/dev\/mmcblk0p/,"", $1); print $1}'` or you can even skip the substitution,

```
current=$(mount | awk '$3 == "/" {print $1}'

# Deduce target rootfs partition based on current rootfs partition number
if [ $current = "/dev/mmcblk0p2" ]; then
    newroot=/dev/mmcblk0p3
...
```

Reply:
@kengsooncheah I like the no sub solution! Updated